src/RoboSaverVirtualModule.sol

```diff
@@ -214,9 +247,16 @@ contract RoboSaverVirtualModule {
         if (_isExternalTxQueued()) revert ExternalTxIsQueued();
 
         if (_action == PoolAction.WITHDRAW) {
-            _poolWithdrawal(CARD, _amount);
+            /// @dev Close the pool in case the $EURe available for withdrawal is less than the deficit
+            uint256 withdrawableEure =
+                BPT_STEUR_EURE.balanceOf(CARD) * BPT_STEUR_EURE.getRate() * (MAX_BPS - slippage) / 1e18 / MAX_BPS;
```

Comment:
for readability my pref in ordering will be:

```suggestion
(BPT_STEUR_EURE.balanceOf(CARD) * BPT_STEUR_EURE.getRate() / 1e18) * (MAX_BPS - slippage) / MAX_BPS;
```

Reply:
i always understood that in solidity best practice is to multiply everything first, then divide. this to minimise weird rounding/precision errors

eg https://ethereum.stackexchange.com/questions/156819/are-multiplications-after-division-unsafe

Reply:
in my suggestion you are still always multiplying that dividing not to encounter the precision loss issue, no?

it is just diff ordering to understand quicker why actually `1e18` is req, instead of having the deduction of `bps - slipp`

Reply:
no, in your suggestion the `/ 1e18` will take place before the `* (MAX_BPS - slippage) / MAX_BPS`

Reply:
the key is to make the number in memory as big as possible first before dividing.

i agree it goes as the cost of readability here, but i wouldnt want to introduce rounding errors for the sake of readability

im open to examples or other references on how to do both at the same time (no errors but readable)

Reply:
that's true. i still think in this case it is just shuffling the order. the output will be the same as you are doing:

`multiplication - division - multiplication - division`

will check in another moment for curiosity. resolving for now

Reply:
<img width="1432" alt="Screenshot 2024-06-03 at 21 38 20" src="https://github.com/onchainification/robosaver/assets/84875062/dc699294-b2d9-44c9-823a-f980761b9a48">

local test of either approach